Assets/Scripts/Weapons/Projectile/Projectile.cs

```diff
@@ -30,23 +32,35 @@ void OnTriggerEnter(Collider other)
         {
             // Damage player
             otherObject.GetComponent<Player>().ChangeHealthBy(-damage);
-            Destroy(gameObject);
+            hit();
         }
     }
 
+    /**
+        Method that is called when the projectile hits something and will be destroyed
+     */
+    void hit()
+    {
+        // Notify any event listeners
+        OnHit(transform.position);
```

Comment:
I don't think this will cause problems for us, is that this may break if there are no listeners. There's a shorthand for invoking events in Unity: `eventName?.Invoke(argumentsIfAny)`. In this case, you'd do `OnHit?.Invoke(transform.position)`. It's essentially shorthand for `if(OnHit != null) OnHit(transform.position)`. This ensures that we don't invoke the event unless there's a listener.

Reply:
Oh right, forgot about that.